ipn/ipnlocal/local.go

```diff
@@ -2215,6 +2221,24 @@ func (b *LocalBackend) setNetMapLocked(nm *netmap.NetworkMap) {
 		b.activeLogin = login
 	}
 
+	// Determine if file sharing is enabled
+	fs := false
+	if nm != nil && nm.SelfNode != nil && len(nm.SelfNode.Capabilities) > 0 {
+		for _, c := range nm.SelfNode.Capabilities {
+			if c == tailcfg.CapabilityFileSharing {
+				fs = true
+				break
```

Comment:
rather than control flow like this, please pull this back to a function similar to how it was, but with a different signature: the function can take a NetworkMap and return a bool. Then you can just "return true" here.

Reply:
I wanted to hide this looping from any other code, to avoid misuse in the future (other code should use either `b.capFileSharing` field directly if locked or `b.hasCapFileSharing()` method if not locked). So I was thinking about implementing it as an anonymous function or simply loop directly.
Anyway, I'll implement it as you've suggested.